src/main.cpp

```diff
@@ -1834,8 +1834,14 @@ bool ConnectBlock(const CBlock& block, CValidationState& state, CBlockIndex* pin
         return true;
     }
 
-    bool fScriptChecks = true;
-    if (fCheckpointsEnabled) {
+    assert(nScriptCheckThreads >= 1);
+    const int64_t timeBarrier = GetTime() - 24 * 3600 * nScriptCheckThreads;
+    // Blocks that have varius days of POW behind them makes them secure in
+    // that actually online nodes checked the scripts, so during a catch-up we
+    // don't need to check the scripts. The block signatures themselves are
+    // still checked.
+    bool fScriptChecks = block.nTime > timeBarrier;
+    if (fScriptChecks && fCheckpointsEnabled) {
```

Comment:
This if () block should be removed -- the hard-coded checkpoints will always be way behind the current time.